apis/filesystem/src/test/java/org/jclouds/filesystem/integration/FilesystemBlobIntegrationTest.java

```diff
@@ -82,26 +79,34 @@ public void testCreateBlobWithExpiry() throws InterruptedException {
       super.testCreateBlobWithExpiry();
    }
 
-   /* Java on OS X does not support extended attributes, which the filesystem backend
-    * uses to implement user metadata */
+   /*
+    * Java on OS X does not support extended attributes, which the filesystem
+    * backend uses to implement user metadata
+    */
    @Override
    protected void checkUserMetadata(Map<String, String> userMetadata1, Map<String, String> userMetadata2) {
-      if (!isMacOSX()) {
-         super.checkUserMetadata(userMetadata1, userMetadata2);
-      }
+      checkExtendedAttributesSupport();
+      super.checkUserMetadata(userMetadata1, userMetadata2);
    }
 
-   /* Java on OS X does not support extended attributes, which the filesystem backend
-    * uses to implement storage tiers */
+   /*
+    * Java on OS X does not support extended attributes, which the filesystem
+    * backend uses to implement storage tiers
+    */
    @Override
    protected void testPutBlobTierHelper(Tier tier, PutOptions options) throws Exception {
-      if (!isMacOSX()) {
-         super.testPutBlobTierHelper(tier, options);
-      }
+      checkExtendedAttributesSupport();
+      super.testPutBlobTierHelper(tier, options);
    }
 
    @Override
    public void testSetBlobAccess() throws Exception {
       throw new SkipException("filesystem does not support anonymous access");
    }
+
+   private void checkExtendedAttributesSupport() {
+      if (isMacOSX()) {
+         throw new SkipException("filesystem does not support extended attributes on Mac OSX");
```

Comment:
If we end up using `SkipException` in lots of places, thoughts on creating a `TestUtils.skip` helper for readability? Then we could write:

```
if (isMacOSX()) {
   skip("filesystem does not support extended attributes on Mac OSX");
}
```
?

Thanks for taking care of this, @nacx!

Reply:
I don't think this adds anything an prevents `grep` from finding tests we skip, part of the expanded scope of this pull request.